### ISSUE TITLE
I LIKE IT!

### ISSUE THREAD

**Issue by PySimpleGUI** [Concrete18/Home-Control-Interface#1]
Long time ago I was wanting to be able to write code that controls my Hue lights.  Back then Ruby was the only thing I could find. 

Now, with Python able to control them and you've added a GUI, LOTS of capabilities open up for many people. Congrats!

Thank you for posting a screenshot!!! Make a HUGE difference.

**Comment by Concrete18** [comment]
Thanks! Any ideas on how to improve it? Also how did you even find this?

P.S. I use PySimpleGUIWX to make a taskbar icon for some other stuff but I don't know how to use threading and classes to allow more control over functions. I know how to technically make clicking the icon to toggle a function and change the icon but I can't thread the class stuff and still have the while loop for the taskbar keep access to the variables. I don't know how else to allow it to communicate.

**Comment by Concrete18** [comment]
Oh! Did you find this because of the requirements.txt?

**Comment by PySimpleGUI** [comment]
I can't disclose all my secrets :-)

If you want to run both a system tray icon AND a window at the same time, then you have to use timeouts on each.  It's a more or less polling situation.  It's far from ideal.

You can't run PySimpleGUI (any of the ports) as a thread, so don't try.  

You can mix PySimpleGUI and PySimpleGUIWx.  I do this often when I have an application that uses both the system tray icon and windows.  I use PySimpleGUIWx only for the tray icon and PySimpleGUI for the window portion.  Both share a single event loop with a timeout on both reads.  Again, not efficient, but it works.

There is a new PySimpleGUI system tray icon, but it doesn't run in the system tray (yet), instead it runs like a little icon on your desktop, just above the system tray.  A consultant has been trying to get a version of another package pstray to work in conjunction with PySimpleGUI so that a true system tray can be used, but it's not done yet.

In summary, use timeouts on your read calls in order to get what looks like multi-threading to work.

**Comment by Concrete18** [comment]
Do you have an example that you have done it one of your repositories? Name
it and I will look to see. I have already gotten threading to work in order
to have the tray icon running with the main function existing within the
thread. There was just no communication without global variables.

Anyway example or not I will try and figure out time outs.

**Comment by Concrete18** [comment]
I see the timeout stuff in the documentation now. I was in a hurry looking for a taskbar and failed to read enough.

**Comment by Concrete18** [comment]
Hey, what do you think of these two projects? They use your tray icon.
https://github.com/Concrete18/Weather-Wallpaper-Changer
and
https://github.com/Concrete18/Standing-Reminder

I did not use timeouts and figured out classes as well!

**Comment by PySimpleGUI** [comment]
I'll give them a try sometime soon...

I missed your earlier question about places to check for code.  It's well documented (in the first 10 pages of the main doc you'll find links to everything including 200 demo programs http://demos.PySimpleGUI.org as well as port specific demos which are on the same GitHub.

I don't remember right now where the reminder app is but I'm pretty sure it's in one of the folders or under a repository in my account.  

Sounds like you've been pretty successful in the past 5 days.  Congrats on making stuff!  I'll give 'em a look.
